sdk/playwrighttesting/microsoft-playwright-testing/src/core/playwrightService.ts

```diff
@@ -114,6 +115,7 @@ const getServiceConfig = (
   if (options?.serviceAuthType === ServiceAuth.ACCESS_TOKEN) {
     // mpt PAT requested and set by the customer, no need to setup entra lifecycle handlers
     validateMptPAT(exitWithFailureMessage);
+    checkTokenExpiryWarning();
```

Comment:
``if (options?.useCloudHostedBrowsers === false) {``

Let's add a function for one time operation before the above line. That function is responsible for checking if the environment variable `ONE_TIME_OPERATION_FLAG` is set or not. If not set, it will call appropriate functions like checkTokenExpiryWarning and then set the above flag.

This will ensure a function like checkTokenExpiryWarning is not responsible for setting one-time operation flags, instead it can focus on it's responsibility of token expiry check and warning,